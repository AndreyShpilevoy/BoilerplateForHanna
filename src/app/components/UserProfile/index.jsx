import React from 'react';
import {string, shape, number} from 'prop-types';

const UserProfile = ({className, userProfile}) => {
    const {id, name} = userProfile;
    const userInfoString = `User id: ${id}, User name: ${name}`;
    return (
        <div className={className}>
            {userInfoString}
        </div>
    );
};

UserProfile.propTypes = {
    className: string,
    userProfile: shape({
        id: number.isRequired,
        name: string.isRequired,
        order: number.isRequired
    }).isRequired
};

UserProfile.defaultProps = {
    className: ''
};

export default UserProfile;
