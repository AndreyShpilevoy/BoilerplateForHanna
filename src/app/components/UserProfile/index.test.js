/* eslint-disable no-undef, max-statements, import/first, react/jsx-filename-extension */

import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UserProfile from './index';

configure({adapter: new Adapter()});

describe('UserProfile', () => {
    it('UserProfile match expected snapshot', () => {
        const userProfile = {id: 1, name: 'some user', order: 1};
        expect(shallow(<UserProfile className='userProfile-class-name' userProfile={userProfile} />)).toMatchSnapshot();
    });
});
