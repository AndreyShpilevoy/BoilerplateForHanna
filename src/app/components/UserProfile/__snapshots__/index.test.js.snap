// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile UserProfile match expected snapshot 1`] = `
<div
  className="userProfile-class-name"
>
  User id: 1, User name: some user
</div>
`;
